### ISSUE TITLE
Wire.requestFrom() should not be followed by Wire.endTransmission().

### ISSUE THREAD

**Issue by Koepel** [sparkfun/SparkFun_GridEYE_Arduino_Library#3]
In the file "SparkFun_GridEYE_Arduino_Library.cpp", in the function `getRegister()` there is a `Wire.endTransmission()` after the `Wire.requestFrom()`. That `Wire.endTransmission()` is currently at line 907 and should be removed.
Explanation: [Common-mistakes#2](https://github.com/Koepel/How-to-use-the-Arduino-Wire-library/wiki/Common-mistakes#2)

When looking at the function `getRegister()`, something is not okay. The parameter `len` is only 1 or 2, so the while-statement to retrieve data with `Wire.read()` is odd. When only one byte is requested, the second call to `Wire.read()` returns -1, that translates to 255 for a byte. That 255 is pushed to the high byte and returned in the `result`. To avoid confusion, it is easier to use `getRegister8()` and `getRegister16()`.

**Comment by galed3e3** [comment]
getRegister() is indeed broken:  isFramerate10FPS() does not work - always returns false, regardless of frame rate - and the only thing isFramerate10FPS() does is use getRegister() to read the frame rate register and then return true or false.  The fix to getRegister() that @Koepel posted [here ](https://github.com/chrisxkeith/ir-qwiic-example/issues/1)  corrects the problem.  His/her code, a direct, drop-in replacement for the current getRegister() is:
```
int16_t GridEYE::getRegister(unsigned char reg, int8_t len)
{
  int16_t result = 0;     // default 0, if the sensor was not connected

  _i2cPort->beginTransmission(_deviceAddress);
  _i2cPort->write(reg);
  _i2cPort->endTransmission(false);    // 'false' for a repeated start

  _i2cPort->requestFrom((uint8_t)_deviceAddress, (uint8_t)len);
  if (_i2cPort->available() == len)    // got valid data ?
  {
    if (len == 1)
    {
      result = (int16_t) _i2cPort->read();
    }
    else if (len == 2)
    {
      int lsb = _i2cPort->read();
      int msb = _i2cPort->read();

      // concat bytes
      result = (int16_t) (msb << 8 | lsb);
    }
  }
  return result;
}
```

**Comment by PaulZC** [comment]
Hi @Koepel & @galed3e3  ,

Apologies for the VERY slow reply!

Yes, correct, the read methods did need some major improvements. I've corrected this in v1.0.2 - which will be released shortly.

Best wishes,
Paul
